posts/2020-09-01-jakarta-microprofile-beta-200010.adoc

```diff
@@ -0,0 +1,355 @@
+
+In the mpHealth-3.0 feature for Open Liberty:
+
+The overall default Readiness status was changed to DOWN, with an empty response until all the deployed application(s) have been started. A new MicroProfile Config property (mp.health.default.readiness.empty.response=UP) is introduced to change the overall default Readiness check status to UP, during application start up, that do not have any user-defined health checks.
+
+The HealthCheckResponseBuilder.state(Boolean UP) method was also renamed to HealthCheckResponseBuilder.status(Boolean UP) for HealthCheckResponse deserialization compatibility, where the JSON health check response string can now be deserialized into an HealthCheckResponse object.
+
+The deprecated @Health qualifier was removed, and you should use the @Liveness or @Readiness qualifiers in your HealthCheck implementations, as appropriate.
+
+Applications are expected to provide health check procedures by implementing the HealthCheck interface with the @Liveness or @readiness annotations. These are used by Open Liberty to verify the Liveness or Readiness of the application, respectively. Add the logic of your health check in the call() method, and return the HealthCheckResponse object, by using the simple up()/down() methods from the API:
+
+[source, java]
+----
+**Liveness Check**
+@Liveness
+@ApplicationScoped
+public class AppLiveCheck implements HealthCheck {
+...
+    @Override
+     public HealthCheckResponse call() {
+       ...
+       HealthCheckResponse.up("my-liveness-check");
+       ...
+     }
+}
+
+**Readiness Check**
+@Readiness
+@ApplicationScoped
+public class AppReadyCheck implements HealthCheck {
+...
+    @Override
+     public HealthCheckResponse call() {
+       ...
+       HealthCheckResponse.named("my-app-readiness").status(isMyAppReady()).build();
+       ...
+     }
+}
+...
+----
+
+To view the status of each health check, access the either the `http://<hostname>:<port>/health/live` or `http://<hostname>:<port>/health/ready endpoints`.
+
+More information: 
+* link:https://github.com/eclipse/microprofile-health/releases/tag/3.0-RC3[MicroProfile Health Check 3.0 Release Page]
+* link:https://github.com/eclipse/microprofile-health/blob/master/spec/src/main/asciidoc/release_notes.asciidoc[Release notes]
+
+[#opentracing]
+=== MicroProfile OpenTracing 2.0
+
+MicroProfile OpenTracing 2.0 can be used to profile and monitor applications built using microservice architecture.
+
+MicroProfile OpenTracing 2.0 has upgraded the OpenTracing API to version 0.33.0.  This allows the 
+use of tracing backends and their libraries that are built on OpenTracing API 0.33.0.
+
+==== Try it now
+
+Include the following in the `server.xml`:
+
+[source, xml]
+----
+    <feature>mpOpenTracing-2.0</feature>
+----
+
+Also configure a tracing backend such as Jaeger or Zipkin 
```

Comment:
missing punctuation at the end of sentence

Reply:
Done